### PR TITLE
coverity/149459: CHECKED_RETURN (false positive)

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: src/nvim/eval.c [hunk 1]
```diff
@@ -7472,7 +7472,9 @@ static void f_and(typval_T *argvars, typval_T *rettv)
 static void f_api_info(typval_T *argvars, typval_T *rettv)
 {
   Dictionary metadata = api_metadata();
-  object_to_vim(DICTIONARY_OBJ(metadata), rettv, NULL);
+  if (!object_to_vim(DICTIONARY_OBJ(metadata), rettv, NULL)) {
```
Comment (bfredl) [inline src/nvim/eval.c]: why not just `(void)object_to_vim(DICTIONARY_OBJ(metadata), rettv, NULL);` ?

Comment (justinmk) [inline src/nvim/eval.c]: will that satisfy coverity?

Comment (justinmk) [inline src/nvim/eval.c]: oh, yes: http://stackoverflow.com/a/8107269/152142

Comment (justinmk) [inline src/nvim/eval.c]: I could also have assigned it to the global `ignored`, but I thought asserting adds a bit more information about invariants.

**Pull request merged**